Sources/ArcGISToolkit/Components/JobManager/JobManager.swift

```diff
@@ -0,0 +1,301 @@
+        let array = jobs.map { $0.toJSON() }
+        UserDefaults.standard.setValue(array, forKey: defaultsKey)
+    }
+    
+    /// Load any jobs that have been saved to User Defaults.
+    private func loadState() {
+        Logger.jobManager.debug("Loading state.")
+        guard let strings = UserDefaults.standard.array(forKey: defaultsKey) as? [String] else {
+            return
+        }
+        
+        jobs = strings.compactMap {
+            try? Job.fromJSON($0) as? any JobProtocol
+        }
+    }
+    
+    /// The current background task identifier.
+    var backgroundTaskIdentifier: UIBackgroundTaskIdentifier?
+    
+    /// Ends any current background task.
+    private func endCurrentBackgroundTask() {
+        guard let backgroundTaskIdentifier else {
+            Logger.jobManager.debug("No current background task to end.")
+            return
+        }
+        Logger.jobManager.debug("Ending current background task.")
+        UIApplication.shared.endBackgroundTask(backgroundTaskIdentifier)
+        self.backgroundTaskIdentifier = nil
+    }
+    
+    /// Starts a background task for extended time in the background if we have jobs that are
+    /// started but have yet to begin polling.
+    private func beginBackgroundTask() {
+        // Jobs that are started but do not yet have a server job ID are jobs that
+        // can benefit from starting a background task for extra execution time.
+        // This will hopefully allow a job to get to the polling state before the app is suspended.
+        // Once in a polling state that's where background refreshes can check job status.
+        if !jobs.contains(where: ({ $0.status == .started && $0.serverJobID.isEmpty })) {
+            Logger.jobManager.debug("No jobs that require starting a background task.")
+            return
+        }
+        
+        // Already started.
+        guard backgroundTaskIdentifier == nil else {
+            Logger.jobManager.debug("Background task already started.")
+            return
+        }
+        
+        Logger.jobManager.debug("Starting a background task.")
+        
+        let identifier = UIApplication.shared.beginBackgroundTask() {
+            Logger.jobManager.debug("Out of background processesing time.")
+            self.endCurrentBackgroundTask()
+        }
+        
+        self.backgroundTaskIdentifier = identifier
+    }
+}
+
+/// An enum that defines a schedule for background status checks.
+public enum BackgroundStatusCheckSchedule {
+    /// No background status checks will be requested.
+    case disabled
+    /// Requests that the system schedule a background check at a regular interval.
+    /// Ultimately it is up to the discretion of the system if that check is run.
+    case regularInterval(interval: TimeInterval)
+}
+
+extension Logger {
+    /// A logger for the job manager.
+    /// To enable logging add an environment variable named "LOGGING_FOR_JOB_MANAGER".
```

Comment:
Similar to the other location, a bit more direction may be useful.

```suggestion
    /// To enable logging add an environment variable named "LOGGING_FOR_JOB_MANAGER" under Scheme
    /// > Arguments > Environment Variables
```